internal/httpproxy/http_proxy_test.go

```diff
@@ -643,3 +646,33 @@ func TestShouldSkipWebSocketRequest(t *testing.T) {
 		})
 	}
 }
+
+func TestIsHealthCheckRequest(t *testing.T) {
+	testCases := []struct {
+		name           string
+		request        *http.Request
+		expectedResult bool
+	}{
+		{
+			name:           "Healthcheck request",
+			request:        httptest.NewRequest(http.MethodGet, healthCheckPath, nil),
+			expectedResult: true,
+		},
+		{
+			name:           "POST request to healthcheck path",
+			request:        httptest.NewRequest(http.MethodPost, healthCheckPath, nil),
+			expectedResult: false,
+		},
+		{
+			name:           "Proxy request",
+			request:        httptest.NewRequest(http.MethodGet, "/api/v1/namespaces/default/pods", nil),
```

Comment:
nit: It doesn't matter for this test but to be precise, a "proxy request" starts with a HTTP CONNECT, not GET (This GET request comes after the CONNECT request).